functions/components/metaTagInjector.ts

```diff
@@ -0,0 +1,41 @@
+type MetaTagInjectorInput = {
+  title: string
+  image?: string
+  url: string
+}
+
+/**
+ * Wrapper class for Cloudflare's HTMLRewriter {@link https://developers.cloudflare.com/workers/runtime-apis/html-rewriter}
```

Comment:
i would rephrase this as it's not really wrapping HTMLRewriter. maybe say it's a "listener" class for HTMLRewriter events. or a "subscriber"